SHARDING.md

```diff
@@ -0,0 +1,254 @@
+
+## Creating Key Material
+
+The public-good instance uses a Tink keyset to sign checkpoints. Tink
+encrypts a signing key using a "key encryption key" managed by GCP KMS,
+and loads the key into memory on service startup.
+
+Tink is beneficial for reliability and cost reduction - KMS is now only
+a startup dependency rather than a runtime dependency, and signing happens
+exclusively in memory.
+
+To create a shard signing key:
+
+1. Grant yourself access to use the new signing key. Update the `staging.tf`
+or `production.tf` config to include:
+
+```
+resource "google_kms_key_ring_iam_member" "rekor-tiles-keyring" {
+  key_ring_id = "<project-rekor or projectsigstore-staging>/<location, e.g. global>/<name of shard keyring>"
+  role        = "roles/cloudkms.cryptoOperator"
+
+  member = "user:<username>@sigstore.dev"
+}
+```
+
+2. Run `terraform plan` and `terraform apply` using the GHA workflow.
+
+3. Using [tinkey](https://developers.google.com/tink/tinkey-overview#installation), create an encrypted Tink keyset:
+
+```
+tinkey create-keyset --key-template ED25519 --out enc-keyset.cfg --master-key-uri gcp-kms://projects/<project name>/locations/<region>/keyRings/<key-ring>/cryptoKeys/<key>
+```
+
+Save the output file `enc-keyset.cfg`, which will be used as a value in the Helm chart.
+
+4. Revoke IAM access by removing `google_kms_key_ring_iam_member`, and run `plan` and `apply`.
+
+## Creating K8s Resources
+
+Update `argocd/bootstrap-utilities/staging/values.yaml` or `argocd/bootstrap-utilities/production/values.yaml`
+to include the new shard. Under `rekor-tiles.shards`, copy the previous shard instantiation, and update
+the values accordingly based on the Terraform resources names. For `keyset`, copy the contents of
+`enc-keyset.cfg` (Since this value is encrypted, it's secure to include it in the config rather than
+use GCP Secrets).
+
+<!--- TODO: Add code pointers for values -->
+
+Merge, and wait for ArgoCD to create the resources and spin up the service.
+
+## Update Prober
+
+Update the prober configuration for
+[staging](https://github.com/sigstore/sigstore-probers/blob/8608ff2cbc6a9121f12c75cb01a18f32e1428195/.github/workflows/prober-staging.yml#L28)
+or
+[prod](https://github.com/sigstore/sigstore-probers/blob/8608ff2cbc6a9121f12c75cb01a18f32e1428195/.github/workflows/prober-prod.yml#L26)
+to specify the new shard name. Merge and kick off the workflow.
+
+<!--- TODO: Update prober to support testing against multiple Rekor shard URLs -->
+<!--- TODO: Update prober to test reads and writes separately, since frozen shards will still support reads -->
+
+## Perform Manual Testing
+
+Using Cosign, test signing and verification against this new shard. You will need to manually update the SigningConfig
+and TrustedRoot files and pass them via CLI flags since those files are pulled from TUF and will not yet have been updated.
+
+<!--- TODO: Update with example TrustedRoot and SigningConfig -->
```

Comment:
include example call to cosign using custom trustedroot and signingconfig.

Reply:
Updated with a curl-based example rather than cosign.